app/helpers/ExerciseConfig/Pipeline/Box/Boxes/JvmCompilationBox.php

```diff
@@ -0,0 +1,168 @@
+
+    /**
+     * Static initializer.
+     * @throws ExerciseConfigException
+     */
+    public static function init()
+    {
+        if (!self::$initialized) {
+            self::$initialized = true;
+            self::$defaultInputPorts = array(
+                new Port((new PortMeta())->setName(self::$ARGS_PORT_KEY)->setType(VariableTypes::$STRING_ARRAY_TYPE)),
+                new Port(
+                    (new PortMeta())->setName(self::$SOURCE_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port(
+                    (new PortMeta())->setName(self::$EXTRA_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port((new PortMeta())->setName(self::$JAR_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)),
+                new Port((new PortMeta())->setName(self::$COMPILER_EXEC_PORT_KEY)->setType(VariableTypes::$STRING_TYPE))
+            );
+            self::$defaultOutputPorts = array(
+                new Port(
+                    (new PortMeta())->setName(self::$CLASS_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                )
+            );
+        }
+    }
+
+    /**
+     * JudgeNormalBox constructor.
+     * @param BoxMeta $meta
+     */
+    public function __construct(BoxMeta $meta)
+    {
+        parent::__construct($meta);
+    }
+
+
+    /**
+     * Get type of this box.
+     * @return string
+     */
+    public function getType(): string
+    {
+        return self::$BOX_TYPE;
+    }
+
+    /**
+     * Get default input ports for this box.
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function getDefaultInputPorts(): array
+    {
+        self::init();
+        return self::$defaultInputPorts;
+    }
+
+    /**
+     * Get default output ports for this box.
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function getDefaultOutputPorts(): array
+    {
+        self::init();
+        return self::$defaultOutputPorts;
+    }
+
+    /**
+     * Get default name of this box.
+     * @return string
+     */
+    public function getDefaultName(): string
+    {
+        return self::$DEFAULT_NAME;
+    }
+
+
+    /**
+     * Compile box into set of low-level tasks.
+     * @param CompilationParams $params
+     * @return array
+     * @throws ExerciseConfigException
+     */
+    public function compile(CompilationParams $params): array
+    {
+        // Create a separate directory where all *.class files will end up.
+        $mkdirTask = new Task();
+        $mkdirTask->setPriority(Priorities::$INITIATION);
+        $mkdirTask->setType(TaskType::$INITIATION);
+        $mkdirTask->setCommandBinary(TaskCommands::$MKDIR);
+        $mkdirTask->setCommandArguments(
+            [
+                ConfigParams::$SOURCE_DIR . $this->getDirectory(
+                ) . ConfigParams::$PATH_DELIM . self::$COMPILATION_SUBDIR
```

Comment:
I do not fancy placing newlines between parentheses representing an empty argument list.

Reply:
Yes... this is weird... autoformatter did this, I will try to repair it